### PR TITLE
Enforce style types for all `style` props.

### PR DESCRIPTION
**Pull request by jeresig**
When working on implementing the new <Layout/> component I realized that I was going to need to have true, enforced, Flow types for the CSS properties that are in Aphrodite and in all of the `style` props.

I've added a new `StyleType` that we should use for all of our `style` props in every component (it's exported from wonder-blocks-core).

Additionally, I've updated the Aphrodite flow rules to match what we're expecting. In the process I confirmed an error that I was seeing when running Jest (that we should've named `::-moz-osx-font-smoothing` to `::MozOsxFontSmoothing`, React didn't like the former). There were a couple small issues like this that I've resolved.

I held off on implementing the exact style props that we want for each component, we can tackle that later, at the very least right now we can enforce that the style props are actually real CSS properties.

Note that I'm not going to land this until `<Layout/>` is ready as I've removed the function option from `<View/>`'s `style` prop. I'll bring that forward in a follow-up PR so that we can evaluate that work separately.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 721cf3a1c83ebb6f589be1d8f5dc057247e8ef0b

https://deploy-preview-262--wonder-blocks.netlify.com

**Comment by kevinbarabash** [comment]
I wasn't sure when this was going to happen b/c it seemed low priority relative to some of the issue we were facing like bugs and publish/deploy issues.  I'm super excited about this!  🎉 

One wrinkle though is that it won't be useful in webapp without some additional work there.  Although we're exporting flow types, we'll need to "teach" webapp how to use them.  This might be as easy as putting some symlinks in `javascript/node_modules` to the right files in `third_party/node_modules` or it may be really hard in which case we'll have to wait until the webpack upgrade.  Hopefully it's the former. 🤞

**Comment by jeresig** [comment]
Yayy! Happy to help :D

And yeah, it'd be great if it worked in Webapp but I agree that it's likely
lower priority. At the very least I needed the Flow types to be consistent
within wonder-blocks itself in order to ensure that the <Layout/> stuff was
going to make sense (and it's going to help us ensure that all of our CSS
typing in our components is accurate, as well).

**Comment by kevinbarabash** [comment]
It'll also help with documentation to provide guidance to people as to what styles they should be using so that there won't be a whole bunch to clean up once we enable this in webapp.

File: packages/wonder-blocks-button/components/button-core.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ const sharedStyles = StyleSheet.create({
         outline: "none",
         textDecoration: "none",
         boxSizing: "border-box",
-        "::-moz-focus-inner": {
+        "::MozFocusInner": {
```
Comment (wishdasher) [inline packages/wonder-blocks-button/components/button-core.js]: For reasons I haven't fully investigated yet, changing it to this doesn't apply the style correctly to the component. https://deploy-preview-262--wonder-blocks.netlify.com/#button-1 on Firefox still has the inner focus ring.

**Comment by kevinbarabash** [comment]
@jeresig I resolved the merge conflicts.  Please double check the fixes before merging.

**Comment by kevinbarabash** [comment]
> we should've named ::-moz-osx-font-smoothing to ::MozOsxFontSmoothing, React didn't like the former

Does `aphrodite` like `::MozOsxFontSmoothing`?

File: flow-typed/aphrodite.flow.js [hunk 2]
```diff
@@ -323,67 +321,51 @@ type CSSProperties = {|
...
+
+    // This is used by wonder-blocks-grid/util/styles.js
+    "@media (min-width: 1168px)"?: _CSSProperties,
 
-    // Hacks
-    // NOTE(jeresig): All of these are hacks and abuses of Aphrodite and
-    // should be removed.
-    ":hover .chevron"?: CSSProperties,
+    // Aphrodite Internal
+    _definition?: _CSSProperties,
```
[the remark this answers is not in the document]
Comment (jeresig) [reply flow-typed/aphrodite.flow.js]: I didn't see an obvious way to do this -- it's technically a property of Aphrodite, it's just internal/hidden...

File: packages/wonder-blocks-modal/__snapshots__/generated-snapshot.test.js.snap [hunk 3]
```diff
@@ -2007,10 +2007,11 @@ exports[`wonder-blocks-modal example 4 1`] = `
             "boxSizing": "border-box",
             "color": "#ffffff",
             "display": "flex",
-            "flex": "1 1 auto",
+            "flex": "1 0 auto",
             "flexDirection": "column",
             "height": "100%",
             "margin": 0,
+            "maxWidth": 392,
```
[the remark this answers is not in the document]
Comment (jeresig) [reply packages/wonder-blocks-modal/__snapshots__/generated-snapshot.test.js.snap]: I'm... not sure!

File: packages/wonder-blocks-tooltip/components/tooltip-tail.js [hunk 4]
```diff
@@ -385,7 +385,7 @@ export default class TooltipTail extends React.Component<Props> {
             <View
                 style={[
                     styles.tailContainer,
-                    offset,
+                    {...offset},
```
[the remark this answers is not in the document]
Comment (jeresig) [reply packages/wonder-blocks-tooltip/components/tooltip-tail.js]: Ah yes - the types on `offset` conflicted with the types in our Aphrodite styles, so making a new object fixes that!

**Pull request merged**